app/controllers/APIController.scala

```diff
@@ -22,9 +26,64 @@ class APIController @Inject() (
     status.asJson.spaces2
   }
 
+  def parseFromJson(request: Request[AnyContent]): Option[ProductFromJson] = {
+
+    Console.println(request.body.asText)
+    val parseResult: Option[ProductFromJson] = {
+      request.body.asText match {
+        case None =>
+          BadRequest("Request was invalid, nothing in request body")
+          None
+        case Some(rawText) => {
+          println("Request body has text")
+          circe.parser.parse(rawText) match { // Is there valid Json?
+            case Left(failure) =>
+              BadRequest("Request was invalid, no valid Json");
+              None
+            case Right(json) =>
+              println(json, "Json matched")
+              parser
+                .decode[ProductFromJson](json.toString()) match { // Can that Json be decoded into a case class?
+                case Left(error) =>
+                  BadRequest(
+                    "Request was invalid, json could not be parsed into case class"
+                  )
+                  None
+                case Right(product) => Some(product)
+              }
+          }
+        }
+      }
+    }
+    println(parseResult, parseResult.getOrElse("Could not get parseResult"))
```

Comment:
The `printlns` need to be cleared up, they are not needed and pollute the space (debugging printlns should always be removed on committed code)